### PR TITLE
Upgraded React Native version and fixed Related Artists layout issue

### PR DESCRIPTION
**Pull request by sarahscott**
- [x] Pretty straightforward RN upgrade to v0.29.0
- [x] Fixed known issue in the new RN version related to `flex-wrap` for related artists
- [x] Rigorously searched for other weirdness in new RN version

simulator screen shot jul 6 2016 6 12 00 pm

With the addition of `alignItems: flex-start`, everything is beautiful again:

simulator screen shot jul 6 2016 6 09 07 pm

Just like they said: https://github.com/facebook/react-native/releases/tag/v0.28.0

### CODE REVIEW

**Comment by alloy** [comment]
Have you tested that layout issue fix on iPad and also with artists where the related artists don’t fully fill the grid and that the remainder is left aligned?

As in, those grid fixes we had to make before when we started testing on iPad.

**Comment by alloy** [comment]
Just found this ticket https://github.com/facebook/react-native/issues/8607. So we should double check if any scrollviews go blank for us.

**Comment by alloy** [comment]
Also need to verify that our nested scrollviews monkey-patches still work as expected. i.e. if the artworks grid is still fetching works when you reach the end of the list.

**Comment by sarahscott** [comment]
This still has the issue where show titles aren't truncating, but otherwise looks good. The grids re-fetch and align properly. I think it's important to get this merged so our subsequent Home view work doesn't run into layout issues later, so I'm making a separate issue for the text truncation.

**Comment by alloy** [comment]
I’ve _suggested_ that [Libraries.io](https://libraries.io/npm/options) lists the license of the `options` package as MIT.

We can’t do much about those deprecated packages, so this is good on my part.

File: lib/relay/config.js [hunk 1]
```diff
@@ -20,11 +18,10 @@ let metaphysicsURL;
...
 Relay.injectNetworkLayer(
-  new Relay.DefaultNetworkLayer(metaphysicsURL, {
-    headers: {
-      'X-ACCESS-TOKEN': Emission.authenticationToken,
-    }
-  })
+  new Relay.DefaultNetworkLayer(metaphysicsURL)
+    // headers: {
+    //   Authorization: 'Basic SECRET'
+    // })
```
Comment (alloy) [inline lib/relay/config.js]: This should not be included.

Comment (alloy) [inline lib/relay/config.js]: Might need a rebase?

Comment (sarahscott) [inline lib/relay/config.js]: Which lines in particular should not be included?

Comment (alloy) [inline lib/relay/config.js]: The auth headers should not be removed.

**Pull request merged**